sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -72,14 +81,17 @@ AZ_NODISCARD AZ_INLINE bool az_span_is_valid(az_span span, int32_t min_length, b
           the length.
   */
   return (
-      (((null_is_valid && (az_span_ptr(span) == NULL) && (span_length == 0) && (span_capacity == 0))
-        || (az_span_ptr(span) != NULL && (span_length >= 0) && (span_capacity >= span_length)))
-       && min_length <= span_length));
+      ((null_is_valid && (ptr + span_length + span_capacity == 0))
```

Comment:
if ptr is NULL, can we do ptr + span_length?  (NULL + constant) ?

If this is changing somehow the logic of this method, we might want to update the docs for it

Reply:
> if ptr is NULL, can we do ptr + span_length? (NULL + constant) ?

I believe so, yes. NULL is the defined as 0.
https://stackoverflow.com/questions/1296843/what-is-the-difference-between-null-0-and-0

> If this is changing somehow the logic of this method, we might want to update the docs for it

It isn't. That said, I will either explicitly add comments explaining the details OR revert it if it hurts readability too much.